### DISCUSSION TITLE
How to Increase Prometheus Storage Retention Time -running prometheus docker container

### DISCUSSION THREAD

**Discussion by petervinoth** [prometheus/prometheus#11850] [Q&A]
Hi,

How to Increase Prometheus Storage Retention Time running prometheus docker container? I don't use config and cmd in docker just i run this "docker run -d --name prometheus -p 9090:9090 -v $PWD/prometheus.yml:/etc/prometheus/prometheus.yml prom/prometheus".

So please let me know how to i change storage retention in running docker container.

Regards,
Vinoth

**Comment by roidelapluie** [comment]
The storage retention time for Prometheus can be set using a command line flag when starting the Prometheus container.

To set the storage retention time to 30 days, you would use the `--storage.tsdb.retention.time` flag when running the docker run command, like this:

`docker run -d --name prometheus -p 9090:9090 -v $PWD/prometheus.yml:/etc/prometheus/prometheus.yml prom/prometheus --storage.tsdb.retention.time=30d`

**Reply by petervinoth** [reply]
@roidelapluie Thanks for your reply. I already run the Prometheus docker container. I missed adding this flag "--storage.tsdb.retention.time". it's default 15 days but now just i need to change 1 year. So how to add the retention flag in running the Prometheus container?
Note: I did not use a docker-compose file.

**Comment by petervinoth** [comment]
any update?

**Comment by jmsfnch** [comment]
Posting here in case it helps someone.... I was also struggling with an updated retention flag.

The original script would work:
```
docker run \
        --name prometheus \
        -p 9090:9090 \
        -v /home/james/git/prometheus-config/prometheus.yml:/etc/prometheus/prometheus.yml \
        -v /home/james/git/prometheus-config/resolv.conf:/etc/resolv.conf \
        -v prometheus-storage:/prometheus \
        prom/prometheus &
```

But appending the flag ```--storage.tsdb.retention.time=365d &``` would return a error when parsing the config file:

```
james@rc-docker1:~/git/prometheus-config$ cat startPrometheus.sh
docker run \
        --name prometheus \
        -p 9090:9090 \
        -v /home/james/git/prometheus-config/prometheus.yml:/etc/prometheus/prometheus.yml \
        -v /home/james/git/prometheus-config/resolv.conf:/etc/resolv.conf \
        -v prometheus-storage:/prometheus \
        prom/prometheus \
        --storage.tsdb.retention.time=365d &
james@rc-docker1:~/git/prometheus-config$
james@rc-docker1:~/git/prometheus-config$ ./startPrometheus.sh
james@rc-docker1:~/git/prometheus-config$ ts=2023-10-21T09:50:07.038Z caller=main.go:487 level=error msg="Error loading config (--config.file=prometheus.yml)" file=/prometheus/prometheus.yml err="open prometheus.yml: no such file or directory"

james@rc-docker1:~/git/prometheus-config$
```


For some reason, the container was now looking in the wrong place for the config ```/prometheus/prometheus.yml```, (as opposed to  the former ``/etc/prometheus/prometheus.yml`` path).

### Solution:

I updated the script again to set command line flags for the config file and data paths; everyhing is working fine, with the correct retention period and paths shown in ``http://host:9090/status``

```
docker run \
        --name prometheus \
        -p 9090:9090 \
        -v /home/james/git/prometheus-config/prometheus.yml:/etc/prometheus/prometheus.yml \
        -v /home/james/git/prometheus-config/resolv.conf:/etc/resolv.conf \
        -v prometheus-storage:/prometheus \
        prom/prometheus \
        --config.file=/etc/prometheus/prometheus.yml --storage.tsdb.path=/prometheus --storage.tsdb.retention.time=365d &
```

**Reply by serakis** [reply]
Thanks, just what I needed!
